README.md

```diff
@@ -67,38 +73,64 @@ plugins:
+    subnets:
+      # tag key
+      - tagKey: <tag_name>
+        # an array of values
+        tagValues:
+          - '<tag_value>'
+    # optional if `subnets` option is specified
+    # list of tag key and value or names
+    securityGroups:
+      # tag key
+      - tagKey: <tag_name>
+        # an array of values
+        tagValues:
+          - '<tag_value>'
+      # optional if `tagKey` and `tagValues` are specified
+      - names:
+        - '<security_group_name>'
 
-# (Optional) set a config for the specific function
+# Optional: Either set `custom.vpcDiscovery` or `functions.<function name>.vpcDiscovery`
 functions:
-  example1:
+  example:
     handler: handler.example
-    # inherit `custom.vpcDiscovery` config if specified
+    # inherit `custom.vpcDiscovery` config in case `custom.vpcDiscovery` is specified
   example2:
     handler: handler.example
-    # skip vpc configuration
+    # skip vpc configuration for the current function
     vpcDiscovery: false
   example3:
     handler: handler.example
-    # inherit `custom.vpcDiscovery` and override security group names
+    # inherit `custom.vpcDiscovery` config in case `custom.vpcDiscovery` is specified and override security group names
     vpcDiscovery:
       vpcName: '${opt:env}'
-      securityGroupNames:
-        - '${opt:env}_<name of security group>'
+      securityGroups:
+        - tagKey: <tag_name>
+          tagValues:
+            - '<tag_value>'
+        ...
   example4:
     handler: handler.example
-    # override basic subnet names and security group names
+    # override or set basic subnets and security groups items
     vpcDiscovery:
       vpcName: '${opt:env}'
-      subnetNames: # optional if securityGroupNames are specified
-        - '${opt:env}_<name of subnet>'
-      securityGroupNames:  # optional if subnetNames are specified
-        - '${opt:env}_<name of security group>'        
+      subnets: # optional if `subnets` option is specified
+        - tagKey: <tag_name>
+          tagValues:
+            - '<tag_value>'
+        ...
+      securityGroups: # optional if `subnets` option is specified
+        - tagKey: <tag_name> # optional if `names` option is specified
+          tagValues: # optional if `names` option is specified
+            - '<tag_value>'
+        - names: # optional if `tagKey` and `tagValues` are specified
+          - '${opt:env}_<name of security group>'
+        ...
 ```
-> NOTE: The naming pattern we used here was building off the vpc name for the subnet and security group by extending it with the the subnet and security group name. This makes it easier to switch to different vpcs by changing the environment variable in the command line.
+> NOTE: The naming pattern we used here was building off the vpc name for the subnet and security group by extending it with the subnet and security group values. This makes it easier to switch to different vpcs by changing the environment variable in the command line.
```

Comment:
We can probably remove this line now that we are supporting arbitrary tags instead of specific names